internal/provider/client.go

```diff
@@ -63,18 +74,41 @@ func (c *AAPClient) GetHosts(stateId string) (*AnsibleHostList, error) {
 	resp, err := client.Do(req)
 
 	if err != nil {
-		return nil, err
+		return -1, []byte{}, err
 	}
 
 	defer resp.Body.Close()
 
 	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return -1, []byte{}, err
+	}
+	return resp.StatusCode, body, nil
+}
+
+func (c *AAPClient) doRequestWithBody(method string, path string, req_data *bytes.Reader) (int, []byte, error) {
```

Comment:
I tried using one function with `nil` but was not working

Reply:
What do you mean by not working? It seems to compile without issues. I'd also suggest using an `io.Reader` instead of a `*bytes.Reader` here and elsewhere. In general, if there's an interface you can use, it's better to use that than a specific implementation of it.

Reply:
done